### ISSUE TITLE
(node) AppSync requests fail with ForbiddenException

### ISSUE THREAD

**Issue by simonbuchan** [aws/aws-sdk-js-v3#1826]
**Describe the bug**

As with title, at least ListGraphqlApis and GetGraphqlApi actions in AppSync fail with ForbiddenException in the v3.0.0 release.

This one is a bit confusing. It seems like AppSync is incorrectly formatting the request or something, straightforward usage is giving ForbiddenExceptions, while they seem to have been working in an pre-release version of v3 (I think as recently as rc.7 worked, but I can't be sure)

Doing a bit of spelunking it *looks* like it's generating a `GET` to `https://appsync.us-west-2.amazonaws.com/apis` with reasonable looking sigv4 signing for ListGraphqlApis, but [the ref](https://docs.aws.amazon.com/appsync/latest/APIReference/API_ListGraphqlApis.html) says it should be `../v1/apis`, so hopefully an easy fix?

**SDK version number**

**Is the issue in the browser/Node.js/ReactNative?**
Only tried in Node.js

**Details of the browser/Node.js/ReactNative version**
v15.2.1

**To Reproduce (observed behavior)**

```console
% node --experimental-repl-await
Welcome to Node.js v15.2.1.
Type ".help" for more information.
> var AppSync = require("@aws-sdk/client-appsync");
undefined
> var appSync = new AppSync.AppSync({ region: "us-west-2" });
undefined
> var res = await appSync.listGraphqlApis({});
Uncaught:
ForbiddenException: Forbidden
    at deserializeAws_restJson1ListGraphqlApisCommandError (/Users/simon/code/admin-tool/node_modules/@aws-sdk/client-appsync/dist/cjs/protocols/Aws_restJson1.js:3578:41)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async /Users/simon/code/admin-tool/node_modules/@aws-sdk/middleware-serde/dist/cjs/deserializerMiddleware.js:7:20
    at async /Users/simon/code/admin-tool/node_modules/@aws-sdk/middleware-signing/dist/cjs/middleware.js:12:24
    at async StandardRetryStrategy.retry (/Users/simon/code/admin-tool/node_modules/@aws-sdk/middleware-retry/dist/cjs/defaultStrategy.js:55:46)
    at async /Users/simon/code/admin-tool/node_modules/@aws-sdk/middleware-logger/dist/cjs/loggerMiddleware.js:7:22 {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 403,
    httpHeaders: {
      'content-type': 'application/json',
      'content-length': '23',
      connection: 'keep-alive',
      date: 'Tue, 22 Dec 2020 04:43:07 GMT',
      'x-amzn-requestid': '8eb07db3-9fd1-4afc-bfd0-762a25127dd3',
      'x-amzn-errortype': 'ForbiddenException',
      'x-amz-apigw-id': 'X7_mQG-ivHcFrVg=',
      'x-cache': 'Error from cloudfront',
      via: '1.1 b026be0d561433ed8842cc9bb94c7f8f.cloudfront.net (CloudFront)',
      'x-amz-cf-pop': 'AKL50-C2',
      'x-amz-cf-id': 'UHFcpXWyWpoJTyQeQJDQ6GoPC_9ywE4oxazI8Adxi0hU7B0yTI62Ig=='
    },
    requestId: '8eb07db3-9fd1-4afc-bfd0-762a25127dd3',
    attempts: 1,
    totalRetryDelay: 0
  }
}
```

Now, in my experience, this isn't normally what you get if you are missing permission, but in case it should be that, I have no issue with other clients or apis (these should both be using my default profile in `~/.aws/credentials`):

```console
% aws --region us-west-2 appsync list-graphql-apis
{
    "graphqlApis": [
        {
... snip
```

**Expected behavior**

I suppose that it works like v2 and the CLI.

**Comment by simonbuchan** [comment]
Found a workaround, confirming the `/v1` is the issue:

```
  const appSync = new AppSync.AppSyncClient({ ... });
  appSync.middlewareStack.add(
    (next) =>
      async (args: any) => {
        args.request.path = "/v1" + args.request.path;
        return await next(args);
      },
    { name: "path fixup", step: "build", priority: "high" },
  );
```

**Comment by AllanZhengYP** [comment]
Hi @simonbuchan 

I can confirm `/v1` exists in the v2 SDK's [JSON model](https://github.com/aws/aws-sdk-js/blob/e04b8184452ba3d77cda8d52a1ee5b61668c9cd8/apis/appsync-2017-07-25.min.json#L854) but missed in the [Smithy model](https://github.com/aws/aws-sdk-js-v3/blob/e7dce39ede5e73b6c60647a189345ff7d6010a69/codegen/sdk-codegen/aws-models/appsync.2017-07-25.json#L3031) used by v3. Thank you for reporting this. 

I will comment here once we have a fix for it.

**Comment by rehrumesh** [comment]
Thanks @simonbuchan for your solution. However I'd love to see this gets fixed soon in the library.

**Comment by hfahlbusch** [comment]
We had the same issue and used @simonbuchan 's fix. (Thanks for that!)

Today I upgraded aws-sdk/client-appsync to 1.98.0 and the fix broke function with an error:
AccessDeniedException: Unable to determine service/operation name to be authorized

Then I removed the fix and it worked without for getGrapqhlIntrospection action.
Maybe this ticket has been fixed with the latest release.

**Comment by simonbuchan** [comment]
Confirming this seems to be fixed in 3.13.1.
